src/overlay/EnvelopeHandler.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+// Copyright 2018 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "overlay/Peer.h"
+#include "transactions/TransactionFrame.h"
+#include "xdr/Stellar-overlay.h"
+
+namespace stellar
+{
+
+struct ItemKey;
+
+class EnvelopeHandler
```

Comment:
`EnvelopeHandler` seems to not be the right interface exposed by overlay: by taking `Peers` as parameters it keeps requiring its consumers (Herder in this case) to keep track of peers - and this doesn't seem to be the right direction

Reply:
this may be caused by the point I raised above: we probably want to split PendingEnvelope and ReadyEnvelope, which should reduce the dependencies on overlay constructs

Reply:
EnvelopeHandler is supposed to be interface exposed by Transport layer and used by Overlay - Overlay knows about Peers, so it is OK. Herder should not know about EnvelopeHandler at all.